.github/workflows/build.yml

```diff
@@ -7,12 +7,79 @@ on:
     branches: [ "master" ]
 
 jobs:
-
-  build:
-
+  code-formatting:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+    - name: Set up node
+      uses: actions/setup-node@v3
+    - name: Install npm packages
+      run: npm ci
+    - name: Run code formatting tests
+      run: ./tests/run-formatting-tests.sh
+  ruby-linting:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+    - name: Build Docker backend
+      run: |
+        cp dev.env .env
+        docker-compose up --build -d backend
+        docker ps -a
+    - name: Run Ruby formatting tests
+      run: 
+        ./tests/run-ruby-linting-tests.sh
+    - name: Display Docker logs on failure
+      if: ${{ failure() }}
+      run: docker-compose logs
+    - name: Clean up Docker
+      run: docker-compose down
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+    - name: Build Docker backend
+      run: |
+        cp dev.env .env
+        docker-compose up --build -d
+        docker ps -a
+        docker-compose run backend rake db:create
+        docker-compose run backend rake db:migrate
+        sleep 30
+    - name: Display Docker logs on failure
+      if: ${{ failure() }}
+      run: docker-compose logs
+    - name: Run unit tests
+      run: ./tests/run-unit-tests.sh
+    - name: Display Docker logs on failure
+      if: ${{ failure() }}
+      run: docker-compose logs
+    - name: Clean up Docker
+      run: docker-compose down
+  production-build:
+    if: ${{ github.event.issue.push }}
+    needs: [code-formatting, ruby-linting, unit-tests]
     runs-on: ubuntu-latest
-
     steps:
-    - uses: actions/checkout@v3
-    - name: Build the Docker image
-      run: docker build . --file Dockerfile --tag my-image-name:$(date +%s)
+    - name: Checkout repo
+      uses: actions/checkout@v3
+    - name: Build Docker
+      run: |
+        cp dev.env .env
+        docker-compose up --build -d
+        docker ps -a
+        docker-compose run frontend npm run dev-build
+    - name: Display Docker logs on failure
+      if: ${{ failure() }}
+      run: docker-compose logs
+    - name: Deploy
```

Comment:
We don't want pull requests to run deploy, but we do want them to do a production build. Does the current script handle that?

Also, let's remove the `deploy` portion from the travis script so they don't fight with each other.

Reply:
We want to build on all PRs, but we only want to deploy on pushes in master.